client/src/redux/index.js

```diff
@@ -637,9 +646,12 @@ export const reducer = handleActions(
       }
     }),
     [actionTypes.submitComplete]: (state, { payload }) => {
-      let submittedchallenges = [{ ...payload, completedDate: Date.now() }];
-      if (payload.challArray) {
-        submittedchallenges = payload.challArray;
+      const { submittedChallenge, savedChallenges = undefined } = payload;
```

Comment:
```suggestion
      const { submittedChallenge, savedChallenges } = payload;
```
If `savedChallenges` isn't in `payload` this variable will be `undefined` automatically.

Reply:
committed locally, will be resolved on next push